android/src/main/kotlin/com/github/fluttercommunity/flutterdownloader/DownloadWorker.kt

```diff
@@ -0,0 +1,187 @@
+                                val dataBuffer = ByteArray(8096)
+                                var bytesRead: Int
+                                while (`in`.read(dataBuffer, 0, 8096).also { bytesRead = it } != -1) {
+                                    if (isStopped) {
+                                        break
+                                    }
+                                    fileOutputStream.write(dataBuffer, 0, bytesRead)
+                                    bytesReceivedTotal += bytesRead
+                                    contentLength?.let {
+                                        val progress = bytesReceivedTotal * 1000 / contentLength
+                                        if(progress != lastProgress) {
+                                            lastProgress = progress
+                                            withContext(Dispatchers.Main) {
+                                                download.progress = progress
+                                            }
+                                            //println("Update: ${progress / 10.0}%")
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                        if (!isStopped) {
+                            // TODO move to final position
+                            /*
+                            val destFile = File(filePath)
+                            dir = destFile.parentFile
+                            if (!dir.exists()) {
+                                dir.mkdirs()
+                            }
+                            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                                Files.move(
+                                    tempFile.toPath(),
+                                    destFile.toPath(),
+                                    StandardCopyOption.REPLACE_EXISTING
+                                )
+                            } else {
+                                tempFile.copyTo(destFile, overwrite = true)
+                                tempFile.delete()
+                            }*/
+                        } else {
+                            Log.v(TAG, "Canceled task for ${download.urlHash}")
+                            withContext(Dispatchers.Main) {
+                                download.status = DownloadStatus.paused
+                            }
+                            return@withContext Result.failure()
+                        }
+                        Log.i(TAG, "Successfully downloaded taskId ${download.urlHash}")
+                        withContext(Dispatchers.Main) {
+                            download.status = DownloadStatus.completed
+                        }
+                        return@withContext Result.success()
+                    } catch (e: Exception) {
```

Comment:
yeah, to me it looks like too much nesting in a single method

Reply:
I aim here a refactoring

Reply:
Fixed